src/main/java/com/sequenceiq/cloudbreak/service/stack/DefaultStackService.java

```diff
@@ -188,26 +189,27 @@ public void updateStatus(Long stackId, StatusRequest status) {
                 }
                 LOGGER.info("Publishing {} event.", ReactorConfig.STACK_STATUS_UPDATE_EVENT);
                 reactor.notify(ReactorConfig.STACK_STATUS_UPDATE_EVENT,
-                        Event.wrap(new StackStatusUpdateRequest(stack.getTemplate().cloudPlatform(), stack.getId(), status)));
+                        Event.wrap(new StackStatusUpdateRequest(stack.cloudPlatform(), stack.getId(), status)));
             }
         }
     }
 
     @Override
-    public void updateNodeCount(Long stackId, Integer scalingAdjustment) {
+    public void updateNodeCount(Long stackId, HostGroupAdjustmentJson hostGroupAdjustmentJson) {
         Stack stack = stackRepository.findOne(stackId);
         MDCBuilder.buildMdcContext(stack);
         if (!Status.AVAILABLE.equals(stack.getStatus())) {
             throw new BadRequestException(String.format("Stack '%s' is currently in '%s' state. Node count can only be updated if it's running.", stackId,
                     stack.getStatus()));
         }
-        if (0 == scalingAdjustment) {
+        if (0 == hostGroupAdjustmentJson.getScalingAdjustment()) {
             throw new BadRequestException(String.format("Requested scaling adjustment on stack '%s' is 0. Nothing to do.", stackId));
         }
-        if (0 > scalingAdjustment) {
-            if (-1 * scalingAdjustment > stack.getNodeCount()) {
-                throw new BadRequestException(String.format("There are %s instances in stack '%s'. Cannot remove %s instances.", stack.getNodeCount(), stackId,
-                        -1 * scalingAdjustment));
+        if (0 > hostGroupAdjustmentJson.getScalingAdjustment()) {
+            if (-1 * hostGroupAdjustmentJson.getScalingAdjustment() > stack.getFullNodeCount()) {
```

Comment:
the adjustment must be compared to the current hostgroup not the full node count